src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -42,7 +42,11 @@
     {{~/unless~}}
     ) {{type}} {{asLowerCamelCase name~}}
     {{~#if isList~}}
-      []
+      [
+        {{~#if_valid_list_size_restriction~}}
+          {{maxLength}}
```

Comment:
I don't think we should output this.  It's nonsense for lists and doesn't really get used for anything in Matter except sizing some buffers.... and I plan to stop doing that too.

For _strings_ on the other hand, this value is in fact meaningful (in terms of limiting how long a string we will allow to be written) and we should consider showing it there.

Reply:
How do I find if something is a string? Is string the only meaningfully limited item here?

I would like to change if_valid_list_size_restriction to to determine the correct place where sizes are actually sane.

Also I saw no sizes anywhere in "structs" so I assume lists have no explicit bound in zap for structs.

Reply:
Technically, in the matter spec, struct fields can also have caps on size.

In practice, I don't know whether struct fields have a maxLength in ZAP.

For finding whether something is a string, `{{#if (isString type)}}`